packages/network-controller/tests/provider-api-tests/shared-tests.ts

```diff
@@ -277,84 +268,25 @@ export function testsForProviderType(providerType: ProviderType) {
-              },
-            });
-
-            const networkId = await withNetworkClient(
-              { providerType },
-              ({ makeRpcCall }) => {
-                return makeRpcCall(request);
-              },
-            );
-
-            expect(networkId).toStrictEqual('0x1');
-          });
-        });
-      });
-
-      describe('eth_signTransaction', () => {
-        it('throws an error', async () => {
-          const address = '0x0000000000000000000000000000000000000000';
-          const dummyTransaction = {
-            from: address,
-            to: address,
-            gasLimit: '21000',
-            maxFeePerGas: '300',
-            maxPriorityFeePerGas: '10',
-            nonce: '0',
-            value: '10000000000',
-          };
-          const request = {
-            method: 'eth_signTransaction',
-            params: [dummyTransaction],
-          };
-          const expectedResult = `Unknown address - unable to sign transaction for this address: "${address}"`;
-
-          await withMockedCommunications({ providerType }, async (comms) => {
-            comms.mockNextBlockTrackerRequest({ blockNumber: '0x1' });
-
-            const promiseForResult = withNetworkClient(
-              { providerType },
-              async ({ makeRpcCall }) => makeRpcCall(request),
-            );
+        it('does not hit the RPC endpoint, instead returning the configured chain id', async () => {
+          const networkId = await withNetworkClient(
+            { providerType: 'custom', customChainId: '0x1' },
+            ({ makeRpcCall }) => {
+              return makeRpcCall({ method: 'eth_chainId' });
+            },
+          );
 
-            await expect(promiseForResult).rejects.toThrow(expectedResult);
-          });
+          expect(networkId).toStrictEqual('0x1');
         });
       });
     });
   });
 
   describe('methods not included in the Ethereum JSON-RPC spec', () => {
-    describe('methods handled by middleware that returns a static result', () => {
-      const notHandledByMiddleware = [
-        { name: 'net_listening', numberOfParameters: 0, result: true },
-        {
-          name: 'web3_clientVersion',
-          numberOfParameters: 0,
-          result: 'ProviderEngine/v16.0.5/javascript',
-        },
-      ];
-      notHandledByMiddleware.forEach(({ name, numberOfParameters, result }) => {
-        describe(`method name: ${name}`, () => {
-          testsForRpcMethodWithStaticResult(name, {
-            providerType,
-            numberOfParameters,
-            result,
-          });
-        });
-      });
-    });
-
     describe('methods not handled by middleware', () => {
       const notHandledByMiddleware = [
+        { name: 'net_listening', numberOfParameters: 0 },
+        // TODO: Methods to add back when we add testing for subscribe middleware
         // { name: 'eth_subscribe', numberOfParameters: 1 },
         // { name: 'eth_unsubscribe', numberOfParameters: 1 },
```

Comment:
We can uncomment these now; we'll have to test these polyfills at a different point in the pipeline anyway (e.g. in the clients).
```suggestion
        { name: 'eth_subscribe', numberOfParameters: 1 },
        { name: 'eth_unsubscribe', numberOfParameters: 1 },
```

Reply:
Yes I think so

Reply:
Okay done here: https://github.com/MetaMask/metamask-extension/pull/18994